### ISSUE TITLE
vue/require-toggle-inside-transition false positives

### ISSUE THREAD

**Issue by 719media** [vuejs/eslint-plugin-vue#2302]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **eslint-plugin-vue version: 9.18.0

**What did you do?**	

The following all result in linting errors (`vue/require-toggle-inside-transition`), but should not be, as a result of https://github.com/vuejs/eslint-plugin-vue/issues/2290

```vue
<transition>
  <div key="hello" />
</transition>
```

```vue
<transition>
  <slot />
</transition>
```

```vue
<transition appear>
  <div>Something</div>
</transition>
```

**What did you expect to happen?**
No lint errors

**What actually happened?**
lint errors

**Comment by ota-meshi** [comment]
Thank you for posting the issue.
They look like bugs. Please consider opening a PR if you want a quick fix.

**Comment by Shyam-Chen** [comment]
```sh
  3:5  error  The element inside `<transition>` is expected to have a `v-if` or `v-show` directive  vue/require-toggle-inside-transition
```

```vue
<template>
  <Transition name="fade">
    <slot></slot>
  </Transition>
</template>
```

**Comment by DevilTea** [comment]
> Thank you for posting the issue. They look like bugs. Please consider opening a PR if you want a quick fix.

Hello Ota Meshi,

I've opened a PR for it 🫡
